mkosi/__init__.py

```diff
@@ -4632,25 +4637,48 @@ def link_output_sshkey(args: MkosiArgs, sshkey: Optional[SomeIO]) -> None:
             _link_output(args, sshkey.name, args.output_sshkey, mode=0o600)
 
 
-def link_output_split_root(args: MkosiArgs, split_root: Optional[SomeIO]) -> None:
+def substitute_uuid(output_path: Path, part_uuid: Optional[uuid.UUID], substitute: bool) -> Path:
+    if not substitute:
+        return output_path
+
+    assert part_uuid
```

Comment:
Since I was just look at this part specifically:
```suggestion
def substitute_uuid(output_path: Path, part_uuid: uuid.UUID) -> Path:
    if not args.substitute:
        return output_path
```
`substitute` is already in `args`, so no reason to pass it separately and if `part_uuid` being `None` is an error, then let's encode it into the signature.

Reply:
It's only an error if substitution is attempted and `part_uuid` is None. The function could be happily called with `part_uuid` set to `None`.